samples/dnn/optical_flow.py

```diff
@@ -59,41 +70,47 @@ def motion_to_color(self, flow):
-    parser = argparse.ArgumentParser(description='Use this script to calculate optical flow using FlowNetv2',
+    parser = argparse.ArgumentParser(description='Use this script to calculate optical flow',
                                      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
     parser.add_argument('-input', '-i', required=True, help='Path to input video file. Skip this argument to capture frames from a camera.')
     parser.add_argument('--height', default=320, type=int, help='Input height')
     parser.add_argument('--width', default=448, type=int, help='Input width')
-    parser.add_argument('--proto', '-p', default='FlowNet2_deploy_anysize.prototxt', help='Path to prototxt.')
-    parser.add_argument('--model', '-m', default='FlowNet2_weights.caffemodel', help='Path to caffemodel.')
+    parser.add_argument('--proto', '-p', default='', help='Path to prototxt.')
+    parser.add_argument('--model', '-m', required=True, help='Path to model.')
     args, _ = parser.parse_known_args()
 
-    if not os.path.isfile(args.model) or not os.path.isfile(args.proto):
-        raise OSError("Prototxt or caffemodel not exist")
+    if not os.path.isfile(args.model):
+        raise OSError("Model does not exist")
+    if args.proto and not os.path.isfile(args.proto):
+        raise OSError("Prototxt does not exist")
 
     winName = 'Calculation optical flow in OpenCV'
     cv.namedWindow(winName, cv.WINDOW_NORMAL)
     cap = cv.VideoCapture(args.input if args.input else 0)
     hasFrame, first_frame = cap.read()
 
-    divisor = 64.
-    var = {}
-    var['ADAPTED_WIDTH'] = int(np.ceil(args.width/divisor) * divisor)
-    var['ADAPTED_HEIGHT'] = int(np.ceil(args.height/divisor) * divisor)
-    var['SCALE_WIDTH'] = args.width / float(var['ADAPTED_WIDTH'])
-    var['SCALE_HEIGHT'] = args.height / float(var['ADAPTED_HEIGHT'])
+    if args.proto:
+        divisor = 64.
+        var = {}
+        var['ADAPTED_WIDTH'] = int(np.ceil(args.width/divisor) * divisor)
+        var['ADAPTED_HEIGHT'] = int(np.ceil(args.height/divisor) * divisor)
+        var['SCALE_WIDTH'] = args.width / float(var['ADAPTED_WIDTH'])
+        var['SCALE_HEIGHT'] = args.height / float(var['ADAPTED_HEIGHT'])
+
+        config = ''
+        proto = open(args.proto).readlines()
+        for line in proto:
+            for key, value in var.items():
+                tag = "$%s$" % key
+                line = line.replace(tag, str(value))
+            config += line
 
-    config = ''
-    proto = open(args.proto).readlines()
-    for line in proto:
-        for key, value in var.items():
-            tag = "$%s$" % key
-            line = line.replace(tag, str(value))
-        config += line
+        caffemodel = open(args.model, 'rb').read()
 
-    caffemodel = open(args.model, 'rb').read()
+        opt_flow = OpticalFlow(caffemodel, var['ADAPTED_HEIGHT'], var['ADAPTED_WIDTH'], bytearray(config.encode()))
+    else:
+        opt_flow = OpticalFlow(args.model, 360, 480)
```

Comment:
Use args.height and args.width

Reply:
The model weights/trained model I acquired takes the input of `360x480` only, that's why I had to hard code these values.

<img width="1089" alt="Screenshot 2024-01-27 at 12 09 44 PM" src="https://github.com/opencv/opencv/assets/68940203/5bc6bc2b-a885-486a-9984-bf4f12cb7886">

Reply:
Do you think I should leave this as it is? Or, should I find a new model or train a new model?

Reply:
I think that 360x480 gives the best accuracy but using lower resolutions might be faster. But yes, let’s keep original shapes and merge this PR.